### PR TITLE
Throw error when non-mod posts to mod-only comm or when URL is blocked

### PR DESCRIPTION
**Pull request by flamingo-cant-draw**
Currently when a post federates to a Lemmy instance, we only check if the comm is mod only and the whether the user is a mod, but don't actually take any action on this info.

Ref #4963

### CODE REVIEW

File: crates/apub/src/objects/post.rs [hunk 1]
```diff
@@ -220,14 +231,16 @@ impl Object for ApubPost {
       None
     };
 
+    let url_blocklist = get_url_blocklist(context).await?;
+
     if let Some(url) = &url {
+      is_url_blocked(url, &url_blocklist)?;
```
Comment (flamingo-cant-draw) [inline crates/apub/src/objects/post.rs:237]: This isn't anything to do with the issue, I just happened to notice this was missing when looking at the code.

**Review by dullbananas — APPROVED** [review]
If there's a lint that would forbid all unused values other than `()`, then we should start using that. (`#[must_use]` would not be an option for `async` `Result` functions)

File: crates/apub/src/objects/post.rs (hunk 1 above)
Comment (Nothing4You) [reply crates/apub/src/objects/post.rs:237]: Isn't URL blocklist enforced only via API intentionally currently?

Simply dropping these posts when receiving them is a bad solution, as the posts will stay up on the originating instance and community mods will never see them to be able to remove them. For example, if sh.itjust.works uses the URL blocklist feature to ban tinyurl.com and a lemm.ee user posts a link to tinyurl.com to the SJW community the post will be visible for all lemm.ee users but the community moderators will never see the post (unless they're located on lemm.ee).

There are already several other cases that can lead to behavior similar to this, but we should probably not add more to that list.

Comment (Nothing4You) [reply crates/apub/src/objects/post.rs:237]: There might be a middle ground similar to how the slur filter works by censoring URLs instead.

Comment (flamingo-cant-draw) [reply crates/apub/src/objects/post.rs:237]: I'm open to alternatives, but currently we do drop posts if hey have a [blocked URL in the post body](https://github.com/LemmyNet/lemmy/blob/128e78f7c22279fec27d3213a05b440cb1f8db59/crates/apub/src/objects/post.rs#L233). This just makes the behaviour consistent.

File: crates/apub/src/objects/post.rs [hunk 2]
```diff
@@ -180,8 +184,15 @@ impl Object for ApubPost {
...
     let community = page.community(context).await?;
     if community.posting_restricted_to_mods {
-      CommunityModeratorView::is_community_moderator(&mut context.pool(), community.id, creator.id)
-        .await?;
+      let is_mod = CommunityModeratorView::is_community_moderator(
+        &mut context.pool(),
+        community.id,
+        creator.id,
+      )
+      .await?;
```
Comment (dessalines) [inline crates/apub/src/objects/post.rs:192]: I'm going to fix this function so that it only returns a result, but I'll do that after I merge this.

**Review by dessalines — APPROVED** [review]

**Pull request merged**